pkg/render/monitor/monitor.go

```diff
@@ -992,3 +1004,28 @@ func allowTigeraPrometheusOperatorPolicy(cfg *Config) *v3.NetworkPolicy {
 		},
 	}
 }
+
+func (mc *monitorComponent) serviceMonitorCalicoKubeControllers() *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{Kind: monitoringv1.ServiceMonitorsKind, APIVersion: MonitoringAPIVersion},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      KubeControllerMetrics,
+			Namespace: common.TigeraPrometheusNamespace,
+			Labels:    map[string]string{"team": "network-operators"},
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Selector:          metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": "calico-kube-controllers"}},
+			NamespaceSelector: monitoringv1.NamespaceSelector{MatchNames: []string{"calico-system"}},
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					HonorLabels:   true,
```

Comment:
What does this do exactly? I don't know that I understand it from the description in prometheus:
```
// HonorLabels chooses the metric's labels on collisions with target labels.
```

Reply:
ServiceMonitorSpec has TargetLabels 
```// TargetLabels transfers labels from the Kubernetes `Service` onto the created metrics.
// All labels set in `selector.matchLabels` are automatically transferred.```
 As discussed, followed the same behaviour  in other service monitors and used HonorLabels as true.